### ISSUE TITLE
Increase worker thread limits in quic streamer runtime

### ISSUE THREAD

**Issue by t-neIson** [solana-labs-community/solana#198]
#### Problem
The original PR (#24535) was merged but caused flakiness in the CI. `test_quic_server_block_multiple_connections` was failing sporadically. #24535 was reverted due to the CI failures.

#### Summary of Changes
This is a resubmit of the original PR along with a fix to the flaky test.

Fixes #

**Comment by sakeridge** [comment]
streamer/src/quic.rs
```rust
            // Wait for a few milliseconds to allow connection close state to
            // propagate from server to the client
            sleep(Duration::from_millis(10)).await;
```
This seems a little brittle. Could we block_on (with timeout) the connection close instead?

**Comment by t-neIson** [comment]
Yes, I am also not a big fan of adding sleep to the tests. Let me find a better way to fix it.

**Comment by sakeridge** [comment]
Thanks! This looks better to me, assuming it's passing consistently in your tight loop
